lib/NUPACK/emscripten/FullEnsemble.cpp

```diff
@@ -331,10 +368,160 @@ std::string getDotParens(bool pseudoknotted, const int seqlength, oneDnaStruct *
     {
        return singlestructure;
     }
-    
+}
 
+void getEnsembleDefect(char* seqChar, char* dotParensStructure, int temperature, bool pseudoknot, bool multiFold, double *returnEnsembleDefect, bool mfeDefect)
```

Comment:
Is there a reason why you pass in returnEnsembleDefect instead of returning it from the function by value? Also is there a reason why you're using an array instead of a struct?